plugin-core/src/main/java/com/salesforce/bazel/eclipse/classpath/BazelClasspathContainer.java

```diff
@@ -315,7 +321,15 @@ public boolean isValid() throws BackingStoreException, IOException, InterruptedE
                 return true;
             }
             EclipseProjectBazelTargets targets = BazelProjectPreferences.getConfiguredBazelTargets(this.eclipseProject.getProject(), false);
-            List<BazelBuildError> details = bazelWorkspaceCmdRunner.runBazelBuild(targets.getConfiguredTargets(), null, Collections.emptyList());
+            List<BazelLabel> labels = targets.getConfiguredTargets().stream().map(t -> new BazelLabel(t)).collect(Collectors.toList());
+            Map<BazelLabel, IProject> labelToProject = new HashMap<>();
+            for (BazelLabel label : labels) {
+                labelToProject.merge(label, this.eclipseProject.getProject(), (k1, k2) -> {
+                    throw new IllegalStateException("Duplicate label: " + label + " - this is bug");
+                });
+            }
```

Comment:
The logic above is copy/pasted from here: https://github.com/salesforce/bazel-eclipse/blob/a7e562214ee307828c3ec675c1937a667a3983d6/plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelBuilder.java#L165. Can you refactor to avoid the duplication please?

Reply:
Will do!